keps/582-preempt-based-on-flavor-order/README.md

```diff
@@ -0,0 +1,377 @@
+			PodSets:     make([]PodSetAssignment, 0, len(requests)),
+			LastState: workload.AssigmentClusterQueueState{
+				LastAssignedFlavorIdx:  make([]map[corev1.ResourceName]int, 0),
+				CohortGeneration:       0,
+				ClusterQueueGeneration: cq.Generation,
+			},
+			Usage: make(workload.FlavorResourceQuantities),
+		}
+		if cq.Cohort != nil {
+			assignment.LastState.CohortGeneration = cq.Cohort.Generation
+		}
+	}
+  ...
+}
+
+func shouldTryNextFlavor(representativeMode FlavorAssignmentMode, flavorFungibility v1beta1.FlavorFungibility, whetherNeedBorrowing bool) bool {
+	policyPreempt := flavorFungibility.WhenCanPreempt
+	policyBorrow := flavorFungibility.WhenCanBorrow
+	if representativeMode == Preempt && policyPreempt == v1beta1.Preempt {
+		return false
+	}
+
+	if representativeMode == Fit && whetherNeedBorrowing && policyBorrow == v1beta1.Borrow {
+		return false
+	}
+
+	if representativeMode == Fit && !whetherNeedBorrowing {
+		return false
+	}
+
+	return true
+}
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[Y] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+#### Unit Tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `pkg/cache`: `2023-8-22` - `82.9%`
+- `pkg/scheduler`: `2023-8-22` - `80.7%`
+- `pkg/webhook`: `2023-8-22` - `71.2%`
+- `pkg/workload`: `2023-8-22` - `54.9%`
+
+#### Integration tests
+
+<!--
+Describe what tests will be added to ensure proper quality of the enhancement.
+
+After the implementation PR is merged, add the names of the tests here.
+-->
+Should_Schedule_When_Fungibility_Is_Set
```

Comment:
Can you describe a little the scenarios that you plan to test?

See for example https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility#integration-tests